storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -132,12 +125,10 @@ static bool reopen_log_files() {
   log.m_log_flags = log_flags;
   log.m_log_uuid = log_uuid;
   log.m_files = std::move(files);
-  auto file = log.m_files.front();
-  log_encryption_read(log, file);
+  auto file = log.m_files.find(desired_lsn);
+  if (file == log.m_files.end()) return false;
+  log_encryption_read(log, *file);
```

Comment:
Better to check db_err of log_encryption_read and return false, because it can fail if the file is removed by the server.  
Also we could only try to read encryption if redo_file is encrypted.